### PR TITLE
[AzureMonitorExporter] Move _OTELRESOURCE_ as first entry

### PR DESCRIPTION
**Pull request by rajkumar-rangaraj**
Test Framework has a bug and adds items in reverse order. This will be fixed in a separate PR as the changes are not related this PR proposal.

### CODE REVIEW

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Demo/Traces/TraceDemo.cs [hunk 1]
```diff
@@ -36,7 +36,7 @@ public TraceDemo(string connectionString, TokenCredential? credential = null)
                             .AddSource(ActivitySourceName)
                             .AddProcessor(new ActivityFilteringProcessor())
                             .AddProcessor(new ActivityEnrichingProcessor())
-                            .SetSampler(new ApplicationInsightsSampler(new ApplicationInsightsSamplerOptions() { SamplingRatio = 0.0F }))
+                            .SetSampler(new ApplicationInsightsSampler(new ApplicationInsightsSamplerOptions() { SamplingRatio = 1.0F }))
```
Comment (TimothyMothra) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Demo/Traces/TraceDemo.cs:39]: may want to do this change as a separate PR.

**Review by TimothyMothra — APPROVED** [review]
LGTM. one minor comment about the Demo project

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Demo/Traces/TraceDemo.cs (hunk 1 above)
Comment (rajkumar-rangaraj) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Demo/Traces/TraceDemo.cs:39]: It's accidentally changed as a part of this PR - https://github.com/Azure/azure-sdk-for-net/pull/36111. As it not in the core part of the code, included the change in this PR.
